modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1764,6 +1764,11 @@ private ObjectDefinition _publishObjectDefinition(
 			long userId, ObjectDefinition objectDefinition)
 		throws PortalException {
 
+		if (objectDefinition.isApproved()) {
+			throw new ObjectDefinitionStatusException(
+				"The object definition is already published");
+		}
```

Comment:
@larissacribeiro could you please change the `ObjectDefinitionStatusException` class to also receive the language key (`messageKey`) `the-object-definition-is-already-published` so the error can be translated?

Something like:

```
public ObjectDefinitionStatusException(String message) {
	super(message);
}

public ObjectDefinitionStatusException(
	String message, String messageKey) {

	super(message);

	_messageKey = messageKey;
}

public String getMessageKey() {
	return _messageKey;
}

private String _messageKey;
```

And then you can change the `ObjectDefinitionStatusExceptionMapper` class to something like:

```
@Override
protected Problem getProblem(
	ObjectDefinitionStatusException objectDefinitionStatusException) {

	return new Problem(
		Response.Status.BAD_REQUEST,
		ObjectExceptionMapperUtil.getTitle(
			_acceptLanguage, null,
			_language, objectDefinitionStatusException.getMessage(),
			objectDefinitionStatusException.getMessageKey()));
}

@Context
private AcceptLanguage _acceptLanguage;

@Reference
private Language _language;
```

Reply:
Sure @carolmariaabb, just did that, thanks!